source/opt/set_spec_constant_default_value_pass.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "set_spec_constant_default_value_pass.h"
+
+#include <cstring>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "def_use_manager.h"
+#include "spirv-tools/libspirv.h"
+#include "type_manager.h"
+#include "types.h"
+#include "util/parse_number.h"
+
+namespace spvtools {
+namespace opt {
+
+namespace {
+using spvutils::NumberType;
+using spvutils::EncodeNumberStatus;
+using spvutils::ParseAndEncodeNumber;
+
+// Given a numeric value in string form and the expected type of the value,
+// parse the string and encode it in a vector of words. If the value is a
+// scalar integer or floating point value, encodes the value in SPIR-V encoding
+// format. If the value is 'false' or 'true', returns a vector with single
+// word with value 0 or 1 respectively. Returns the vector containing the
+// encoded value on success. Otherwise returns an empty vector.
+std::vector<uint32_t> ParseDefaultValueStr(const char* text,
+                                           const analysis::Type* type) {
+  std::vector<uint32_t> result;
+  if (!strncmp(text, "true", 4)) {
```

Comment:
changed to `strcmp()`. Also updated the blurb to mention: "null-terminated c string" in 0d108d8